switchboard/app/switchboard.hoon

```diff
@@ -0,0 +1,471 @@
+      :~
+        :*
+          %pass  /reject-poke/[uuid]
+          %agent  [peer.call.u.call-state %switchboard]
+          %poke  [%switchboard-reject !>(uuid)]
+        ==
+      ==
+    ==
+  ==
+:: watches
+:: - /incoming/[dap] for agents to register to receive calls
+:: - /call/[uuid] agents will be informed of call UUIDs at the /incoming
+::   path and should subscribe here for SDP messages from the peer
+:: - /call-peer/[uuid] switchboard instances cross-subscribe to each
+::   other's call-peer paths for a call. Leaving represents hanging up
+::   or dropping the call.
+:: - /uuid convenience for web daps, have urbit generate a UUID
+::
+:: See the pokes comment for how agents send SDP messages to switchboard
+:: for delivery to a peer
+++  on-watch 
+  |=  =path
+  ^-  (quip card _this)
+  ?+  path  (on-watch:default path)
+    ::
+    [%incoming @tas ~]
+  =/  dap  +<.path
+  `this(reachable.state (~(put by reachable.state) dap (add 1 (~(gut by reachable.state) dap 0))))
+    ::
+    [%call @ta ~]
+    =/  uuid  +<.path
+    =/  call-state  (~(get by calls.state) uuid)
+    ?~  call-state
+      ~|  "No call with UUID {<uuid>}"  !!
+    ?+  connection-state.u.call-state
+        ~|  "Connection {<uuid>} in state {<connection-state.u.call-state>} cannot take /call watch"  !!
+        :: We've already been rung by a remote switchboard, which is
+        :: waiting on our dap to answer
+        %incoming-ringing
+      =^  cards  state
+      (move-to-state:helper call.u.call-state %connecting)
+      [cards this]
+        :: We just got the %call poke and now the watch by the app
+        %starting
+      =^  cards  state
+      (move-to-state:helper call.u.call-state %dialing)
+      [cards this]
+    ==
+    ::
+    [%call-peer @ta ~]
+    =/  uuid  +<.path
+    =/  call-state  (~(get by calls.state) uuid)
+    ?~  call-state
+      ~|  "No call with UUID {<uuid>}"  !!
+    ?+  connection-state.u.call-state 
+      ~|  "Connection {<uuid>} in state {<connection-state.u.call-state>} cannot-take /call-peer watch"  !!
+        ::
+        %ringing
+      =^  cards  state
+      (move-to-state:helper call.u.call-state %answered)
+      [cards this]
+        ::
+        %connecting
+      =^  cards  state
+      (move-to-state:helper call.u.call-state %connected)
+      [cards this]
+    ==
+    ::
+    [%uuid ~]
+    :_  this
+    :~
+      :*
+        %give  %fact
+        ~[/uuid]
+        %switchboard-uuid  !>((uuid:switchboard-lib bowl))
+      ==
+      :*
+        %give  %kick
+        ~[/uuid]
+        ~
+      ==
+    ==
+  ==
+:: handle hanging up by kicking the call counterparty
+++  on-leave
+  |=  =path
+  ^-  (quip card _this)
+  ?+  path  (on-leave:default path)
+      ::
+      [%incoming @tas ~]
+    =/  dap  +<.path
+    =/  ct  (~(got by reachable.state) dap)
+    ?:  (gte 1 ct)
+      `this(reachable.state (~(del by reachable.state) dap))
+    `this(reachable.state (~(put by reachable.state) dap (dec ct)))
+      ::
```

Comment:
this comment (and all the ones like it that are on the same indentation as the following case statement) should be dedented by two